torch/cuda/jiterator.py

```diff
@@ -0,0 +1,88 @@
+import torch
+from torch import Tensor
+from typing import Callable
+
+import re
+
+__all__ = ["create_jit_fn"]
+
+class _CodeParser:
+    def __init__(self, code_string: str):
+        optional_ws = r"\s*"
+        required_ws = r"\s+"
+        template_params = r"(?P<template_params>\<.+\>)"
+        return_type = r"(?P<return_type>\w+)"
+        function_name = r"(?P<function_name>\w+)"
+        function_params = r"(?P<function_params>\(.+\))"
+        function_body = r"(?P<function_body>\{.+\})"
+
+        pattern = \
+            optional_ws \
+            + "template" \
+            + optional_ws + template_params \
+            + optional_ws + return_type \
+            + required_ws + function_name \
+            + optional_ws + function_params \
+            + optional_ws + function_body \
+            + optional_ws
+
+        result = re.match(pattern, code_string)
+
+        if result is None:
+            raise Exception("Couldn't parse code, please check correctness: ", code_string)
+
+        self.template_params = result["template_params"]
+        self.return_type = result["return_type"]
+        self.function_name = result["function_name"]
+        self.function_params = result["function_params"]
+        self.function_body = result["function_body"]
+
+
+def create_jit_fn(code_string: str, **kwargs) -> Callable:
+    """
+    Create a jiterator-generated cuda kernel.
+
+    Args:
+        code_string (string): CUDA code string to be compiled by jiterator
+        kwargs (Dict, optional): Keyword arguments for generated function
+
+    Examples:
+        >>> code_string = "template <typename T> T my_kernel(T x, T y, T alpha) { return  -x + alpha * y; }"
```

Comment:
This example is very good. Additional details about the form the CUDA string must take would be helpful. Like, it's a templated function that accepts ...

Reply:
Done.